contracts/Unipool.sol

```diff
@@ -135,8 +154,102 @@ contract Unipool is LPTokenWrapper {
         lastUpdateTime = block.timestamp;
         periodFinish = block.timestamp.add(DURATION);
 
-        tradedToken.safeTransferFrom(msg.sender, address(this), _amount);
+        rewardToken.safeTransferFrom(msg.sender, address(this), _amount);
 
         emit RewardAdded(_amount);
     }
+    
+    function reinvestReward() public updateReward(msg.sender) onlyTradable() {
```

Comment:
I wouldn't use a modifier unless it's required in multiple places, if it was then it would increase clarity, but in this situation I think it's unnecessary. Also `onlyTradable` doesn't describe clearly what it's restricting. `isReinvestable` would make more sense, but either way I think you can remove it.